contracts/chain-adapters/SP1_Adapter.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./interfaces/AdapterInterface.sol";
+
+contract DataStore {
+    error NotHubPool();
+    mapping(address => bytes) public dataForTarget;
```

Comment:
Should change this to bytes[] so we don't overwrite the calldata for each target address each time